src/widgets/search-box/search-box.js

```diff
@@ -287,6 +320,59 @@ function addDefaultAttributesToInput(placeholder, input, query, cssClasses) {
   CSSClassesToAdd.forEach(cssClass => input.classList.add(cssClass));
 }
 
+function addReset(input, reset, {reset: resetTemplate}, clearFunction) {
+  reset = {
+    cssClasses: {},
+    template: resetTemplate,
+    ...reset,
+  };
+
+  const resetCSSClasses = {root: cx(bem('reset'), reset.cssClasses.root)};
+  const templateData = {cssClasses: resetCSSClasses};
+
+  let stringNode;
+
+  if (isString(resetTemplate)) {
+    stringNode = Hogan.compile(resetTemplate).render(templateData);
+  }
+
+  if (isFunction(resetTemplate)) {
+    stringNode = resetTemplate(templateData);
+  }
+
+  const htmlNode = createNodeFromString(stringNode);
```

Comment:
We can add more checking for sure, followed the existing code. Will add something more robust, but since we document the options it should not be a problem if it's throw?